relay/channel/task/hailuo/adaptor_test.go

```
@@ -1,16 +1,410 @@
+}
+
+func TestH3QueryUsesV2PathAndEscapesTaskID(t *testing.T) {
+	service.InitHttpClient()
+	requestPath := make(chan string, 1)
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		requestPath <- r.URL.EscapedPath()
+		w.WriteHeader(http.StatusOK)
+	}))
+	t.Cleanup(server.Close)
+
+	_, err := (&TaskAdaptor{}).FetchTask(server.URL, "test-key", map[string]any{
```

golangci-lint (2.13.2)

[error] response body must be closed

(bodyclose)